rimage/transform/parallel_projection.go

```diff
@@ -95,5 +97,149 @@ func (pp *ParallelProjection) PointCloudToRGBD(cloud pointcloud.PointCloud) (*ri
+	minX := math.Max(meanX-float64(sigmaLevel)*stdevX, meta.MinX)
+	maxZ := math.Min(meanZ+float64(sigmaLevel)*stdevZ, meta.MaxZ)
+	minZ := math.Max(meanZ-float64(sigmaLevel)*stdevZ, meta.MinZ)
+
+	// Change the max and min values to ensure the robot marker can be represented in the output image
+	var robotMarker spatialmath.Pose
+	if ppRM.robotPose != nil {
+		robotMarker = *ppRM.robotPose
+		maxX = math.Max(maxX, robotMarker.Point().X)
+		minX = math.Min(minX, robotMarker.Point().X)
+		maxZ = math.Max(maxZ, robotMarker.Point().Z)
+		minZ = math.Min(minZ, robotMarker.Point().Z)
+	}
+
+	// Calculate the scale factors
+	widthScaleFactor := float64(imageWidth-1) / (maxX - minX)
+	heightScaleFactor := float64(imageHeight-1) / (maxZ - minZ)
+
+	// Add points in the pointcloud to a new image
+	im := rimage.NewImage(imageWidth, imageHeight)
+	cloud.Iterate(0, 0, func(pt r3.Vector, data pointcloud.Data) bool {
+		j := (pt.X - minX) * widthScaleFactor
+		i := (pt.Z - minZ) * heightScaleFactor
+		x, y := int(math.Round(j)), int(math.Round(i))
+
+		// Adds a point to an image using the value to define the color. If no value is available,
+		// the default color of white is used.
+		if x >= 0 && x < imageWidth && y >= 0 && y < imageHeight && data != nil {
+			var c rimage.Color
+			if data.HasValue() {
+				c = getProbabilityColorFromValue(data.Value())
+			} else {
+				c = rimage.NewColor(255, 255, 255)
+			}
+			im.Circle(image.Point{x, y}, voxelRadius, c)
+		}
+		return true
+	})
+
+	// Add a red robot marker to the image
+	if ppRM.robotPose != nil {
+		robotMarkerPoint := image.Point{
+			X: int(math.Round((robotMarker.Point().X - minX) * widthScaleFactor)),
+			Y: int(math.Round((robotMarker.Point().Z - minZ) * heightScaleFactor)),
+		}
+		robotMarkerColor := rimage.NewColor(255, 0, 0)
+		im.Circle(robotMarkerPoint, voxelRadius, robotMarkerColor)
+	}
+	return im, nil, nil
+}
+
+// RGBDToPointCloud is unimplemented and will produce an error.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) RGBDToPointCloud(
+	img *rimage.Image,
+	dm *rimage.DepthMap,
+	crop ...image.Rectangle,
+) (pointcloud.PointCloud, error) {
+	return nil, errors.New("converting a RGB image to Pointcloud is current unimplemented for this projection")
+}
+
+// ImagePointTo3DPoint is unimplemented and will produce an error.
+func (ppRM *ParallelProjectionOntoXZWithRobotMarker) ImagePointTo3DPoint(pt image.Point, d rimage.Depth) (r3.Vector, error) {
+	return r3.Vector{}, errors.New("converting an image point to a 3D point is current unimplemented for this projection")
+}
+
+// getProbabilityColorFromValue returns an RGB color value based on the the probability value and defined hit and miss
+// thresholds
+// TODO (RSDK-1705): Once probability values are available this function should be changed to produced desired images.
+func getProbabilityColorFromValue(v int) rimage.Color {
+	var r, g, b uint8
+
+	prob := float64(v) / 100.
```

Comment:
[q] Do you assume that the value `v` is in the form of percentages?

If so, would it make sense to document that in the function description and enforce values to be a percentage value within `[0, 100]` and throw an error if they're not?

The reasons why I think that this function should strictly enforce percentage values are:
1. The behavior of the function would be more predictable. The user knows exactly what it does and is empowered to decide how to crop/filter/edit the data they feed into this function based on their needs.
2. If "invalid" data is fed into it, it doesn't get silently ignored. We'll be able to debug things more on the code level, rather than wondering why the produced images are not what we'd expect them to be.

Reply:
I was a bit torn on how to treat this. The function is is implemented to be used by SLAM where we will either assume no data (orb) or a percentage (carto). However I wasn't sure how strict to impose this definition by erroring out as I could see it also being used by point clouds down the line which have less defined values. 

So my question is:
1. Do we thrown an error if it is outside the range
2. Do we log that it is not a percentage and default to using white if not
3. Do we use potential color information from the pointcloud to define the color of the point in the projection?

Reply:
This is pretty SLAM-specific due to the robot marker. I would err on the side of making this strict for now, and it can always be relaxed later if used for other things.

Reply:
Added error out + test for this case